wnfs/src/public/link.rs

```diff
@@ -12,27 +13,69 @@ use crate::Link;
 /// A link to another node in the WNFS public file system. It can be held as a simple serialised CID or as a reference to the node itself.
 ///
 /// The public file system is a DAG so we don't have to worry bout cyclic references.
-pub type PublicLink = Link<PublicNode>;
+#[derive(Debug, Clone, PartialEq)]
+pub struct PublicLink(Link<PublicNode>);
```

Comment:
I'm in favor of going with `Link<PublicNode>` in use sites instead of `PublicLink`.
This makes code just ever so slightly shorter, but I'm not sure whether it's worth it.
Not a deal-breaker at all! Just throwing it out there. You decide :)

Reply:
We won't be able to specialize, if we need to. Like with `with_dir`. Probably not great examples